src/vivarium_cluster_tools/psimulate/results/processing.py

```diff
@@ -62,16 +64,20 @@ def write_results_batch(
         unwritten_results[batch_size:],
     )
     metadata_to_write = _concat_metadata(existing_metadata, new_metadata_to_write)
-    results_to_write = _concat_results(existing_results, new_results_to_write)
+    results_to_write = _concat_batch_results(new_results_to_write)
 
     start = time.time()
-    # write out updated metadata and results
+    # Write results incrementally as separate batch files per metric
+    batch_timestamp = int(time.time() * 1000)
     for metric, df in results_to_write.items():
-        _safe_write(df, output_paths.results_dir / f"{metric}.parquet")
+        metric_dir = output_paths.results_dir / metric
+        metric_dir.mkdir(exist_ok=True)
+        _safe_write(df, metric_dir / f"batch_{batch_timestamp}.parquet")
```

Comment:
Now the size of the output files will always match the size of the batch. This is probably fine most of the time, but is it really true that we want to conflate these two values?